src/interpolation/details/ArborX_InterpDetailsCompactRadialBasisFunction.hpp

```diff
@@ -0,0 +1,88 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+#define ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX::Interpolation
+{
+
+namespace Details
+{
+
+template <typename T, std::size_t N>
+KOKKOS_INLINE_FUNCTION T evaluatePolynomial(T const x, T const (&coeffs)[N])
+{
+  T eval = 0;
+  for (std::size_t i = 0; i < N; i++)
+    eval = x * eval + coeffs[i];
+  return eval;
+}
+
+} // namespace Details
+
+namespace CRBF
+{
+
+#define CRBF_DECL(NAME)                                                        \
+  template <std::size_t>                                                       \
+  struct NAME;
+
+#define CRBF_DEF(NAME, N, FUNC)                                                \
+  template <>                                                                  \
+  struct NAME<N>                                                               \
+  {                                                                            \
+    template <typename T>                                                      \
+    KOKKOS_INLINE_FUNCTION static constexpr T evaluate(T const y)              \
+    {                                                                          \
+      T const x = Kokkos::min(Kokkos::abs(y), T(1));                           \
```

Comment:
This needs a comment. And I wonder from the efficiency point of view, whether it is faster to simply return 0 instead of evaluating the polynomial for |y| > 1.